airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py

```diff
@@ -77,7 +77,7 @@ def create_from_stream(
                 partition_generator=StreamPartitionGenerator(
                     stream,
                     message_repository,
-                    SyncMode.full_refresh if isinstance(cursor, NoopCursor) else SyncMode.incremental,
+                    SyncMode.full_refresh if isinstance(cursor, FinalStateCursor) else SyncMode.incremental,
```

Comment:
This is a very weird situations as NoopCursor is still probably a `full_refresh` only cursor.

I think there are a couple of contributing factors to this being weird:
* we didn't add proper interface and rely on knowing what is the implementation
* I think it was assumed that `SyncMode` would become irrelevant soon so agreed for a quick patch/tech debt like this but until we remove it properly and ensure that specific sources don't use it through `Source.stream_slices`, I'm not sure we have a good solution for that...

For now, I would propose to rename `FinalStateCursor` as `NoopCursor` and we should be good (probably?)

Reply:
Yeah I think now that we're doing just a flat replacement it should work as the NoopCursor used to decide sync mode. Which yes is probably going away if everything is an incremental in the future. I don't think line will need to change since Noop is gone entirely

Reply:
noting that since NoopCursor is gone and we just have FinalStateCursor, this should be addressed